app/services/api_cost_tracker.py

```
@@ -0,0 +1,879 @@
+            # Get cost by provider
+            cost_by_provider = {}
+            for provider in APIProvider:
+                if period == "daily":
+                    provider_cost = float(await redis.get(f"cost:provider:{provider.value}:{date_key}") or 0)
+                else:
+                    provider_cost = 0.0
+                    for day in range(1, 32):
+                        try:
+                            day_key = now.replace(day=day).strftime("%Y-%m-%d")
+                            provider_cost += float(await redis.get(f"cost:provider:{provider.value}:{day_key}") or 0)
+                        except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)